src/Limits/CoProduct.lidr

```diff
@@ -0,0 +1,35 @@
+> module Limits.CoProduct
+>
+> import Basic.Category
+>
+> %access public export
+> %default total
+>
+> record CoProduct (cat : Category) (a : obj cat) (b : obj cat) where
+>   constructor MkCoProduct
+>   payload: obj cat
+>   inl: mor cat a payload
+>   inr: mor cat b payload
+>   cfgMorphism:
```

Comment:
for the `Product` definition we are working on, we defined a type like

```
record CommutingMorphism (cat : Category)
                         (x : obj cat) (a : obj cat) (b : obj cat) (carrier : obj cat)
                         (pi1 : mor cat carrier a) (pi2 : mor cat carrier b)
                         (f : mor cat x a) (g : mor cat x b) where
  constructor MkCommutingMorphism
  challenger         : mor cat x carrier
  commutativityLeft  : compose cat x carrier a challenger pi1 = f
  commutativityRight : compose cat x carrier b challenger pi2 = g
```

to describe morphisms which commutes with the projections. What do you think? Could it be a good idea to use a similar approach also here?

Reply:
Definitely more readable.